### PR TITLE
Enable nullability in ToolStripPanelRenderEventArgs

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in ToolStripPanelRenderEventArgs.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6422)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripPanelRenderEventArgs.cs [hunk 1]
```diff
@@ -13,9 +11,9 @@ public class ToolStripPanelRenderEventArgs : EventArgs
         /// <summary>
         ///  This class represents all the information to render the toolStrip
         /// </summary>
-        public ToolStripPanelRenderEventArgs(Graphics g, ToolStripPanel toolStripPanel)
+        public ToolStripPanelRenderEventArgs(Graphics g, ToolStripPanel? toolStripPanel)
         {
-            Graphics = g;
+            Graphics = g.OrThrowIfNull();
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripPanelRenderEventArgs.cs:16]: how come tests calling with `null `are passing here? @gpetrou  can you check if there is inconsistency in tests where they are calling with null for `Graphics `?

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripPanelRenderEventArgs.cs [hunk 2]
```diff
@@ -27,7 +25,7 @@ public ToolStripPanelRenderEventArgs(Graphics g, ToolStripPanel toolStripPanel)
         /// <summary>
         ///  Represents which toolStrip was affected by the click
         /// </summary>
-        public ToolStripPanel ToolStripPanel { get; }
+        public ToolStripPanel? ToolStripPanel { get; }
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripPanelRenderEventArgs.cs:28]: same here. Would this be null apart from tests?

Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/ToolStripPanelRenderEventArgs.cs:28]: Similar to https://github.com/dotnet/winforms/pull/6391#discussion_r778588402

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripPanelRenderEventArgs.cs (hunk 1 above)
Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/ToolStripPanelRenderEventArgs.cs:16]: In which tests are you referring to?

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripPanelRenderEventArgs.cs (hunk 2 above)
Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/ToolStripPanelRenderEventArgs.cs:28]: I will make this non-nullable.

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripPanelRenderEventArgs.cs (hunk 1 above)
Comment (dreddy-work) [reply src/System.Windows.Forms/src/System/Windows/Forms/ToolStripPanelRenderEventArgs.cs:16]: @gpetrou , `ToolStripPanelRenderEventArgs_TestData` is passing graphics `null` and is consumed by `ToolStripRenderer_DrawToolStripPanelBackground_Invoke_CallsRenderToolStripPanelBackground`.

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/ToolStripPanelRenderEventArgs.cs:16]: I don't see any null Graphics there. Only ToolStripPanelRenderEventArgs is null which is similar to what happens in many other tests.

**Review by RussKie — APPROVED** [review]

**Comment by RussKie** [comment]
Docs: https://github.com/dotnet/docs/issues/26720

**Pull request merged**